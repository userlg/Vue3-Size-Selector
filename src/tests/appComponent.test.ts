import { describe, it, expect, vi } from "vitest";

import { mount } from "@vue/test-utils";

import AppVue from "../App.vue";

describe("AppVue Test", () => {

    it("App component render properly", async () => {

        const wrapper = mount(AppVue);

        expect(wrapper.html()).toMatchSnapshot();

        expect(wrapper.find("h3").text()).toEqual("M");

    });

    it("Show message h2 selector", async () => {

        const wrapper = mount(AppVue);

        expect(wrapper.find('h2').text()).toEqual('Choose your size');
    });

    it("Show component sizeSelector properly", async () => {
        const wrapper = mount(AppVue);

        expect(wrapper.findComponent({ name: 'SizeSelector' })).toBeTruthy();
    });
    
    it("Show component sizeSelector properly", async () => {
       
        const wrapper = mount(AppVue);
       
        expect(wrapper.findComponent({ name: 'SizeSelector' })).toBeTruthy();
    });

    it("Test on Updated", async () => {
        
        const wrapper = mount(AppVue);

        const buttons = wrapper.findComponent({ name: 'SizeSelector' }).findAll('button');

        await buttons[0].trigger('click');

        expect(wrapper.findComponent('onUpdated')).toBeTruthy();

    });
});