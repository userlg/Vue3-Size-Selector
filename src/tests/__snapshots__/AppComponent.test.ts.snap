// Vitest Snapshot v1

exports[`AppVue Test > App component render properly 1`] = `
"<div class=\\"mx-auto p-4\\"><img src=\\"/public/shirt.png\\" alt=\\"\\" class=\\"h-[150px] block mx-auto\\"></div>
<div class=\\"bg-amber-200 p-10 mx-auto rounded border border-black shadow-md shadow-gray-500 w-full sm:w-3/4 lg:w-1/2 max-w-md\\">
  <div class=\\"flex flex-col justify-around items-center gap-4\\">
    <h2 class=\\"sm:text-2xl font-bold text-center text-xl font-RockSalt capitalize\\"> Choose your size </h2>
    <div class=\\"bg-white rounded p-2 w-20 border border-black\\">
      <h3 class=\\"text-amber-700 text-center font-extrabold text-2xl\\">M</h3>
    </div>
    <div class=\\"flex flex-row gap-1 sm:gap-2 items-center justify-center mx-auto\\"><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
        <h3 class=\\"text-center text-xs sm:text-base\\">XS</h3>
      </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
        <h3 class=\\"text-center text-xs sm:text-base\\">S</h3>
      </button><button class=\\"active bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
        <h3 class=\\"text-center text-xs sm:text-base\\">M</h3>
      </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
        <h3 class=\\"text-center text-xs sm:text-base\\">L</h3>
      </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
        <h3 class=\\"text-center text-xs sm:text-base\\">XL</h3>
      </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
        <h3 class=\\"text-center text-xs sm:text-base\\">XXL</h3>
      </button></div>
  </div>
</div>"
`;
